seq2science/rules/configuration_workflows.smk

```diff
@@ -139,3 +140,47 @@ if config.get("contrasts"):
             f"\nCould not parse DESeq2 contrast '{contrast}'.\n"
             "A DESeq2 design contrast must be in the form '(batch+)column_target_reference'. See the docs for examples.\n")
         _,_,_,_ = parse_contrast(contrast, samples, check=True)
+
+
+def get_read_length(sample):
```

Comment:
Do the inputs need to be checkpoints? Would it not be faster to add these qc files/`rules.get_effective_genome_size.output` as input for macs2? My concern is speed here. If that is not applicable, then nvm :)

Reply:
The read length is also used by e.g. the chromap aligner, which I'll implement now that we finally have working checkpoints. I don't think checkpoints have a big performance hindrance.

Also if we don't use checkpoints on estimation of the read lengths, we would have to estimate the genome size for each sample separately, which is the whole reason i did this :)